tools/build_api.py

```diff
@@ -426,20 +430,27 @@ def merge_region_list(region_list, destination, notify, padding=b'\xFF'):
             notify.info("  Filling region %s with %s" % (region.name, region.filename))
             part = intelhex_offset(region.filename, offset=region.start)
             part.start_addr = None
-            part_size = (part.maxaddr() - part.minaddr()) + 1
+
+            # make same assumption as in region builder; first segment must fit.
+            # this is only to get a neat ToolException anyway, since IntelHex.merge will
+            # throw IntelHex.AddressOverlapError if there's overlapping
+            part_size = 0
+            for start, stop in part.segments():
+                part_size += stop - start
+            merged.merge(part)
+
             if part_size > region.size:
```

Comment:
If we want to keep this check, we should use something better than part_size here. Perhaps we need to duplicate the "last segment before the end of ROM" finding that is used in config?